lldb/include/lldb/Symbol/TypeSystem.h

```diff
@@ -122,6 +106,9 @@ class TypeSystem : public PluginInterface,
   virtual CompilerType DeclGetFunctionArgumentType(void *opaque_decl,
                                                    size_t arg_idx);
 
+  virtual std::vector<lldb_private::CompilerContext>
+  DeclGetCompilerContext(void *opaque_decl);
```

Comment:
I would rather not put any limitations on TypeSystem plug-ins in case they can be more efficient if they can put off parsing information about a type until accessors are called on the `Decl` objects.

Reply:
yea makes sense, looks like that's what all the other TypeSystem APIs do too